### PR TITLE
update the preUpdate process for vertex_ai_index

### PR DESCRIPTION
**Pull request by JoyceYingZhu**
This PR is change the preUpdate process for vertex_ai_index so that when metadata.0.contents_delta_uri has change only first layer mask will apply and no other mask will add.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
Update the preupdate process to set update mask that supported by vertex_index API.

```

### CODE REVIEW

File: mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ resource "google_vertex_ai_index" "index" {
   }
   region   = "us-central1"
   display_name = "tf-test-test-index%{random_suffix}"
-  description = "index for test (updated)"
+  description = "index for test"
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go:142]: Why did you change this back to the unupdated text? Doesn't changing this exercise the update code?

Comment (JoyceYingZhu) [reply mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go:142]: Yes, this is according the change on preupdate process. When field contents_delta_uri update, no other field should update. So in this case field content_delta_uri is updated and description will not update. You can see that in preUpdate we rewrite updateMask when the field content_delta_uri change.

Comment (slevenick) [reply mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go:142]: But a user will likely attempt to update contents_delta_uri alongside other fields. We can't have the resource fail in that case

Comment (JoyceYingZhu) [reply mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go:142]: In that case we apply updateMask=metadata in preupdate so that the other fields will not be updated.

Comment (slevenick) [reply mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go:142]: But that will end up in a two-stage apply for the user, where they run `terraform apply` which succeeds but hasn't fully completed the update it said it would.

If they can't be updated simultaneously then we need to call the API twice, once to update the contents_delta_uri and once to update the rest of the fields.

Why can't we update the entire resource at the same time? Why does the API only allow updating this field by itself?

Comment (JoyceYingZhu) [reply mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go:142]: https://cloud.google.com/vertex-ai/docs/vector-search/update-rebuild-index#update-batch, we mark it in user guideline that don't update the field with other fields.

Comment (slevenick) [reply mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_index_test.go:142]: Sure, but that's not an acceptable way for this resource to work in Terraform. We need to build the resource so a user can update multiple fields at once.

My question is around the design of the API. Why is it that users can't update the field with other fields?

**Review by slevenick — CHANGES_REQUESTED** [review]
Previous review questions still apply
